### ISSUE TITLE
Release GIL

### ISSUE THREAD

**Issue by angloyna** [angloyna/test-conda-bot#6]
I believe that there is a period during execution when Conda does not release the GIL.  

This doesn't affect the common case user and so isn't a priority, but I'm trying to solve multiple conda environments in parallel and would find this convenient.  As a workaround I'm happy to use processes, but if this is easy to do it might be worth doing as a matter of hygiene.  Feel free to close as low priority though if there is a lot more going on.
